### PR TITLE
doc: Add documentation about the path of an event through relay

### PR DESCRIPTION
**Pull request by untitaker**
Run `make docserver` locally to get a nice view, but the github md renderer works too.

EDIT: Temporary deployment: https://getsentry.github.io/semaphore/architecture/ingest-event-path/

### CODE REVIEW

File: docs/architecture/ingest-event-path.md [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+cannot send HTTP requests fully asynchronously (PHP and certain serverless
+platorms), at the cost of status codes being inaccurate. Note that we still
+emit accurate outcomes into Kafka if configured to do so.
+
+## The background task
+
+The HTTP response is out, with a status code that is just a best-effort guess
+at what the actual outcome of the event is going to be. The rest of what used
+to happen synchronously in the Python `StoreView` is done asynchronously, but
+still in the same process.
```
Comment (RaduW) [inline docs/architecture/ingest-event-path.md]: Maybe slightly more clear would be: 

The HTTP response returned is just a best-effort guess at what the actual outcome of the event is going to be. We only return a 4XX code if we know that the response will fail (based on cached information), if we don't we return a 200 and continue to process the event asynchronously. This asynchronous  processing used to happen synchronously in the Python implementation of `StoreView`.  

The effect of this is that the server will respond much faster that before but we might return 200 for events that will ultimately not be accepted.

**Review by RaduW — APPROVED** [review]

**Review by RaduW — APPROVED** [review]

File: docs/architecture/ingest-event-path.md [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+## Processing enabled vs not?
+
+Relay can run as part of a Sentry installation, such as within `sentry.io`'s
+infrastructure, or next to the application as a forwarding proxy. A lot of
+steps described here are skipped or run in a limited form when Relay is *not*
+running in processing mode:
+
+*  Event normalization does different (less) things.
+
+*  In `static` mode, project config is read from disk instead of fetched from
```
Comment (tonyo) [inline docs/architecture/ingest-event-path.md]: Could we perhaps link to the modes description here?
Also, is "processing" also a mode, or more like, a feature? If the former, we should probably add it to the modes list.

**Review by tonyo — CHANGES_REQUESTED** [review]
I like it 👍 
A quick comment about relay modes.

File: docs/architecture/ingest-event-path.md (hunk 2 above)
Comment (untitaker) [reply docs/architecture/ingest-event-path.md]: I want to keep this out of this document. I want to remove some of these modes once the original mode (fetching from sentry) is GA.

**Comment by untitaker** [comment]
We have graphs now!

**Comment by tonyo** [comment]
Nice graphs, yo!

**Review by tonyo — APPROVED** [review]
🏄

**Pull request merged**